File: backend/app/schemas/emergency.py
"""CAZZ SHIELD — Emergency Schemas"""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class EmergencyStatusResponse(BaseModel):
    emergency_mode: bool
    activated_at: Optional[datetime] = None
    activated_by: Optional[str] = None
    reason: Optional[str] = None
    affected_agents: int
    frozen_budgets: int
    revoked_permissions: int
    actions_since_activation: list[dict]


class EmergencyActivateRequest(BaseModel):
    reason: str
    scope: str = "fleet"  # fleet, department, agent
    target_id: Optional[str] = None
    actions: list[str] = ["stop_agents", "freeze_budgets", "revoke_permissions"]


class EmergencyDeactivateRequest(BaseModel):
    reason: str
    restore_agents: bool = True
    restore_budgets: bool = True
    restore_permissions: bool = True


class EmergencyActionResponse(BaseModel):
    action: str
    status: str
    affected_count: int
    timestamp: datetime
    details: Optional[dict] = None

File: frontend/src/lib/emergency.ts
import { api } from './api';

export interface EmergencyStatus {
  emergency_mode: boolean;
  activated_at: string | null;
  activated_by: string | null;
  reason: string | null;
  affected_agents: number;
  frozen_budgets: number;
  revoked_permissions: number;
  actions_since_activation: Array<{
    action: string;
    timestamp: string;
    by: string;
  }>;
}

export interface EmergencyActivateRequest {
  reason: string;
  actions: string[];
}

export interface EmergencyDeactivateRequest {
  reason: string;
  restore_agents: boolean;
  restore_budgets: boolean;
}

export const emergencyApi = {
  getStatus: async () => {
    const response = await api.get<EmergencyStatus>('/emergency/status');
    return response.data;
  },

  activate: async (request: EmergencyActivateRequest) => {
    const response = await api.post('/emergency/activate', request);
    return response.data;
  },

  deactivate: async (request: EmergencyDeactivateRequest) => {
    const response = await api.post('/emergency/deactivate', request);
    return response.data;
  },

  stopAgent: async (agentId: string) => {
    const response = await api.post(`/emergency/agent/${agentId}/stop`);
    return response.data;
  },

  pauseDepartment: async (department: string) => {
    const response = await api.post(`/emergency/department/${department}/pause`);
    return response.data;
  },
};
